RadixWallet/RadixConnect/RadixConnect/RTC/PeerConnection/PeerConnectionNegotiator.swift

```diff
@@ -188,6 +193,24 @@ extension PeerConnectionNegotiator {
 
 		_ = try await onConnectionEstablished.collect()
 		_ = try await onDataChannelReady.collect()
+
+		@Dependency(\.p2pLinksClient) var p2pLinkClient
+		@Dependency(\.jsonEncoder) var jsonEncoder
+
+		let (privateKey, isNewPrivateKey) = try await p2pLinkClient.getP2PLinkPrivateKey(p2pLink.publicKey)
+		let hashedMessageToSign = p2pLink.connectionPassword.messageToHash.hash().data
+		let linkClientInteractionResponse = try P2P.ConnectorExtension.Request.LinkClientInteractionResponse(
+			discriminator: .linkClient,
+			publicKey: HexCodable32Bytes(data: privateKey.publicKey.compressedRepresentation),
+			signature: HexCodable(data: privateKey.signature(for: hashedMessageToSign))
+		)
+		try await peerConnectionClient.sendData(jsonEncoder().encode(linkClientInteractionResponse))
+		log("Sent LinkClientInteractionResponse")
+
+		if isNewPrivateKey {
+			try await p2pLinkClient.storeP2PLinkPrivateKey(p2pLink.publicKey, privateKey)
+		}
```

Comment:
Can this be extracted in a separate function? would help to make the whole negotiation function more readable :)